posthog/api/team.py

```diff
@@ -394,13 +390,15 @@ def perform_destroy(self, team: Team):
             ignore_conflicts=True,
         )
 
+        # TRICKY: We pass in Team here as otherwise the access to "current_team" can fail if it was deleted
+        report_user_action(self.request.user, f"team deleted", team=team)
```

Comment:
This started causing issues after removing the permission as the permission would lazy load "current_team" before the deletion. Hard bug to track down but either way felt it was better to move the reporting here to be explicit and avoid that issue in the future

Reply:
Looks like a mypy error right now